ldpd/ldpd.c

```diff
@@ -1441,6 +1441,13 @@ merge_ifaces(struct ldpd_conf *conf, struct ldpd_conf *xconf)
 				ldpe_if_exit(iface);
 				break;
 			case PROC_LDE_ENGINE:
+				if (iface->ipv4.enabled)
+					lde_route_update_release(iface,
+					    AF_INET);
+				if (iface->ipv6.enabled)
+					lde_route_update_release(iface,
+					    AF_INET6);
```

Comment:
You also need to handle the case where LDP is disabled on a single address family but is still enabled on the other.

Also, with these changes the `"no mpls ldp"` command is no longer working correctly (at least on my dual-stack test topology). Commands like `"show mpls table"` and `"show ip route"` still display the previously learned labels after LDP is unconfigured. When that happens, the log file constains several instances of this error message: `ldpe_dispatch_lde: cannot find neighbor`. No idea why this is happening but it needs to be fixed.

Reply:
I have pushed a new version of code that fixes the address family issues you mentioned.   The show commands display the correct values.    The "ldpe_dispatch_lde: cannot find neighbor" logs occurs because lde is sending label mapping updates to all neighbors  through ldpe when we are disabling ldp on an interface.   Once ldpe has deleted the nbr on the interface, we will see this error message.     This error message occurred before any of these changes were done.